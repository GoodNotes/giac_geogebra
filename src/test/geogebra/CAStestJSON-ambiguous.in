[[[ggbevalarg:=tan((pi)/(2))],[ggbevalans:=regroup(normal(ggbevalarg))],[ggbevalans2:=regroup(ggbevalarg)]], when(length(""+ggbevalans)<=length(""+ggbevalans2),ggbevalans,ggbevalans2)][1]
convert([[0,-1],[1,0]]*(point(2,3)),25)
{ggbtmpvara, ggbtmpvarb, ggbtmpvarc} intersect {ggbtmpvarc, ggbtmpvard}
ggbsort([[[ggbcsans:=0/0],[ggbcsans:=(x)^(2)=-(3)],[ggbvars:=lname(ggbcsans)]],regroup(csolve((x)^(2)=-(3),when(size(ggbvars)==1,ggbvars[0],x)))][1])
ggbsort([[[ggbcsans:=0/0],[ggbcsans:=(x)^(2)=-(1)],[ggbvars:=lname(ggbcsans)]],regroup(csolve((x)^(2)=-(1),when(size(ggbvars)==1,ggbvars[0],x)))][1])
ggbsort(regroup(csolve({(((-(180))*((x)^(2)))+((16)*((y)^(2))))+((1260)*(x))=2025, ((-(48))*((x)^(2)))+((y)^(2))=-(48), (((-(132))*((x)^(2)))+((64)*((y)^(2))))-((924)*(x))=1089},{x, y})))
ggbsort(regroup(csolve({(y)^(2)=(x)-(1), x=((2)*(y))-(1)},{x, y})))
ggbsort(regroup(csolve((z)^(2)=(ggbtmpvark)*((ggbtmpvara)^(2)),z)))
ggbsort(regroup(csolve(((x)^(2))+(1)=0,x)))
[[ggbnumans:=[[ggblimvans:=?],[ggblimvans:=limit((((tan(x))^(2))*((1)-(sin(x))))/(x),(1570796326/1000000000))], [ggblimvans:=when(ggblimvans==inf || ggblimvans==-inf || ggblimvans==undef,ggblimvans,regroup(ggblimvans))],ggblimvans][3]],when(dim(lname(ggbnumans))==0 || lname(ggbnumans)==[unicode0176u],evalf(ggbnumans,30),evalf(regroup(normal(ggbnumans)),30))][1]
[[ggbderivarg0:=Beta(2, x)], regroup(diff(ggbderivarg0, when(length(lname(ggbderivarg0))==0,x,when(count_eq(x,lname(ggbderivarg0))==0,lname(ggbderivarg0)[0],x))))][1]
[[ggbderivarg0:=(x)*(Beta(ggbtmpvara, ggbtmpvarb, x))], regroup(diff(ggbderivarg0, when(length(lname(ggbderivarg0))==0,x,when(count_eq(x,lname(ggbderivarg0))==0,lname(ggbderivarg0)[0],x))))][1]
when ( type({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}) == DOM_SYMBOLIC && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC , (assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},x))-1] , when ( type({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}) == DOM_IDENT && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC && {67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))} == 'x', (assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}=0,x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}=0,x))-1] ,when ( size({ggbtmpvarc, ggbtmpvara}) == 1,flatten1(ggbsort(normal([op(solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},{ggbtmpvarc, ggbtmpvara}))]))),ggbsort(normal([op(solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},{ggbtmpvarc, ggbtmpvara}))])) ) ) )
when ( type({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}) == DOM_SYMBOLIC && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC , (assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},x))-1] , when ( type({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}) == DOM_IDENT && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC && {((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))} == 'x', (assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}=0,x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}=0,x))-1] ,when ( size({ggbtmpvarc, ggbtmpvara}) == 1,flatten1(ggbsort(normal([op(solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},{ggbtmpvarc, ggbtmpvara}))]))),ggbsort(normal([op(solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},{ggbtmpvarc, ggbtmpvara}))])) ) ) )