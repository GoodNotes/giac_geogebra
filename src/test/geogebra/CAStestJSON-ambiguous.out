1>> caseval("init geogebra")
1<< "geogebra mode on"
2>> fractionalPart(x):=sign(x)*(abs(x)-floor(abs(x)))
2<<  (x)->sign(x)*(abs(x)-floor(abs(x)))
3>> is3dpoint(x):=when(size(x[1])==3 && subtype(x[1])==20,true,false)
3<<  (x)->when((((size(x[1]))==3)) && (((subtype(x[1]))==20)),true,false)
4>> xcoord(a):=when(type(a)==DOM_IDENT,xcoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][0],real(a[1])),when(a[0]=='=',coeff(a[1]-a[2],x,1),a[0])))
4<<  (a)->when(((type(a))==identifier),xcoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[0],real(a[1])),when(((a[0])=='='),coeff(a[1]-(a[2]),x,1),a[0])))
5>> ycoord(a):=when(type(a)==DOM_IDENT,ycoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][1],im(a[1])),when(a[0]=='=',coeff(a[1]-a[2],y,1),a[1])))
5<<  (a)->when(((type(a))==identifier),ycoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[1],im(a[1])),when(((a[0])=='='),coeff(a[1]-(a[2]),y,1),a[1])))
6>> zcoord(a):=when(type(a)==DOM_IDENT,zcoordsymb(a),when(a[0]=='pnt',when(is3dpoint(a),a[1][2],0),when(length(a)<3 && a[0] != '=',0,when(a[0]=='=',coeff(a[1]-a[2],z,1),a[2]))))
6<<  (a)->when(((type(a))==identifier),zcoordsymb(a),when(((a[0])=='pnt'),when(is3dpoint(a),(a[1])[2],0),when(((length(a))<3) && ((a[0])<>'='),0,when(((a[0])=='='),coeff(a[1]-(a[2]),z,1),a[2]))))
7>> degasin(x):=normal(asin(x)/pi*180)*unicode0176u
7<<  (x)->normal(asin(x)/pi*180)*unicode0176u
8>> degacos(x):=normal(acos(x)/pi*180)*unicode0176u
8<<  (x)->normal(acos(x)/pi*180)*unicode0176u
9>> degatan(x):=normal(atan(x)/pi*180)*unicode0176u
9<<  (x)->normal(atan(x)/pi*180)*unicode0176u
10>> degatan2(y,x):=normal(arg(x+i*y)/pi*180)*unicode0176u
10<<  (y,x)->normal(arg(x+ί*y)/pi*180)*unicode0176u
11>> ggbabs(x):=when(x[0]=='pnt' || (type(x)==DOM_LIST && subtype(x)==27),l2norm(x),abs(x))
11<<  (x)->when((((x[0])=='pnt')) || ((((type(x))==vector)) && (((subtype(x))==27))),l2norm(x),abs(x))
12>> ggb_is_zero(x):=when(x==0,true,when(type(x)=='DOM_LIST',max(flatten({x,0}))==min(flatten({x,0}))&&min(flatten({x,0}))==0,when(x[0]=='=',lhs(x)==0&&rhs(x)==0,false)))
12<<  (x)->when((x==0),true,when(((type(x))==vector),(((max(flatten({x,0})))==(min(flatten({x,0}))))) && (((min(flatten({x,0})))==0)),when(((x[0])=='='),(((lhs(x))==0)) && (((rhs(x))==0)),false)))
13>> caseval(regroup([[[ggbevalarg:=tan((pi)/(2))],[ggbevalans:=regroup(normal(ggbevalarg))],[ggbevalans2:=regroup(ggbevalarg)]], when(length(""+ggbevalans)<=length(""+ggbevalans2),ggbevalans,ggbevalans2)][1]))
13<< ?
14>> caseval(regroup(convert([[0,-1],[1,0]]*(point(2,3)),25)))
14<< (-3,2)
15>> caseval(regroup({ggbtmpvara, ggbtmpvarb, ggbtmpvarc} intersect {ggbtmpvarc, ggbtmpvard}))
15<< {ggbtmpvarc}
16>> caseval(regroup(ggbsort([[[ggbcsans:=0/0],[ggbcsans:=(x)^(2)=-(3)],[ggbvars:=lname(ggbcsans)]],regroup(csolve((x)^(2)=-(3),when(size(ggbvars)==1,ggbvars[0],x)))][1])))
16<< {x=(ί*√3),x=((-ί)*√3)}
17>> caseval(regroup(ggbsort([[[ggbcsans:=0/0],[ggbcsans:=(x)^(2)=-(1)],[ggbvars:=lname(ggbcsans)]],regroup(csolve((x)^(2)=-(1),when(size(ggbvars)==1,ggbvars[0],x)))][1])))
17<< {x=(ί),x=(-ί)}
18>> caseval(regroup(ggbsort(regroup(csolve({(((-(180))*((x)^(2)))+((16)*((y)^(2))))+((1260)*(x))=2025, ((-(48))*((x)^(2)))+((y)^(2))=-(48), (((-(132))*((x)^(2)))+((64)*((y)^(2))))-((924)*(x))=1089},{x, y})))))
18<< {{x=(19/14),y=(6/7*√55)},{x=(19/14),y=(-6/7*√55)}}
19>> caseval(regroup(ggbsort(regroup(csolve({(y)^(2)=(x)-(1), x=((2)*(y))-(1)},{x, y})))))
19<< {{x=(1-2*ί),y=(1-ί)},{x=(1+2*ί),y=(1+ί)}}
20>> caseval(regroup(ggbsort(regroup(csolve((z)^(2)=(ggbtmpvark)*((ggbtmpvara)^(2)),z)))))
20<< {z=(-√(ggbtmpvara^2*ggbtmpvark)),z=(√(ggbtmpvara^2*ggbtmpvark))}
21>> caseval(regroup(ggbsort(regroup(csolve(((x)^(2))+(1)=0,x)))))
21<< {x=(ί),x=(-ί)}
22>> caseval(regroup([[ggbnumans:=[[ggblimvans:=?],[ggblimvans:=limit((((tan(x))^(2))*((1)-(sin(x))))/(x),(1570796326/1000000000))], [ggblimvans:=when(ggblimvans==inf || ggblimvans==-inf || ggblimvans==undef,ggblimvans,regroup(ggblimvans))],ggblimvans][3]],when(dim(lname(ggbnumans))==0 || lname(ggbnumans)==[unicode0176u],evalf(ggbnumans,30),evalf(regroup(normal(ggbnumans)),30))][1]))
22<< 0.318309886343627006293367151662
23>> caseval(regroup([[ggbderivarg0:=Beta(2, x)], regroup(diff(ggbderivarg0, when(length(lname(ggbderivarg0))==0,x,when(count_eq(x,lname(ggbderivarg0))==0,lname(ggbderivarg0)[0],x))))][1]))
23<< (Gamma(x)*Gamma(x+2)*Psi(x)-Gamma(x)*Gamma(x+2)*Psi(x+2))/Gamma(x+2)^2
24>> caseval(regroup([[ggbderivarg0:=(x)*(Beta(ggbtmpvara, ggbtmpvarb, x))], regroup(diff(ggbderivarg0, when(length(lname(ggbderivarg0))==0,x,when(count_eq(x,lname(ggbderivarg0))==0,lname(ggbderivarg0)[0],x))))][1]))
24<< x^ggbtmpvara*(-x+1)^(ggbtmpvarb-1)+Beta(ggbtmpvara,ggbtmpvarb,x)
25>> caseval(regroup(when ( type({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}) == DOM_SYMBOLIC && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC , (assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},x))-1] , when ( type({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}) == DOM_IDENT && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC && {67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))} == 'x', (assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}=0,x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))}=0,x))-1] ,when ( size({ggbtmpvarc, ggbtmpvara}) == 1,flatten1(ggbsort(normal([op(solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},{ggbtmpvarc, ggbtmpvara}))]))),ggbsort(normal([op(solve({67000=(ggbtmpvarc)*((ggbtmpvara)^(2007)), 3=(ggbtmpvarc)*((ggbtmpvara)^(9))},{ggbtmpvarc, ggbtmpvara}))])) ) ) )))
25<< {{ggbtmpvarc=(3/(67000/3)^(1/222)),ggbtmpvara=((67000/3)^(1/1998))}}
26>> caseval(regroup(when ( type({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}) == DOM_SYMBOLIC && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC , (assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},x))-1] , when ( type({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}) == DOM_IDENT && type({ggbtmpvarc, ggbtmpvara}) == DOM_SYMBOLIC && {((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))} == 'x', (assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}=0,x))[size(assume({ggbtmpvarc, ggbtmpvara}),solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))}=0,x))-1] ,when ( size({ggbtmpvarc, ggbtmpvara}) == 1,flatten1(ggbsort(normal([op(solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},{ggbtmpvarc, ggbtmpvara}))]))),ggbsort(normal([op(solve({((67/10))*((10)^(9))=(ggbtmpvarc)*((ggbtmpvara)^(2007)), (3)*((10)^(8))=(ggbtmpvarc)*((ggbtmpvara)^(950))},{ggbtmpvarc, ggbtmpvara}))])) ) ) )))
26<< {{ggbtmpvarc=(300000000/((67/3)^(1/1057))^950),ggbtmpvara=((67/3)^(1/1057))}}
